### DISCUSSION TITLE
problem to integrate Motorola Hubble MBP85C

### DISCUSSION THREAD

**Discussion by pilot1981** [koush/scrypted#1090] [Q&A]
Hi,
I'm trying to add Motorola Hubble MBP85C to scrypted but I'm unable.

I hacked webcam and I obtained a stable RTSP TCP flow that I can see in Home Assistant using RTSP-TCP integration or using this VLC command:

vlc 'rtsp://user:[pass@192.168.4.20](mailto:pass@192.168.4.20):6668/blinkhd' --network-caching=1000 --rtsp-tcp

If I try to add in scrypted I obtained this logs:

linux x64 #1 SMP PREEMPT_DYNAMIC Wed Aug 16 18:11:54 UTC 2023
server version: 0.50.0
plugin version: @scrypted/rtsp 0.0.54
full
########################
9/18/2023, 7:39:03 PM
########################
[Rebroadcast Plugin]: prebuffer sessions starting in 5 seconds
[Rebroadcast Plugin]: prebuffer sessions releasing if started
[Rebroadcast Plugin]: prebuffer sessions starting in 5 seconds
[Rebroadcast Plugin]: prebuffer sessions releasing if started
[Rebroadcast Plugin]: prebuffer sessions starting in 5 seconds
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
########################
9/18/2023, 7:45:55 PM
########################
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[WebRTC Plugin]: signalingState have-remote-offer
[WebRTC Plugin]: iceGatheringState gathering
[WebRTC Plugin]: signalingState stable
[WebRTC Plugin]: connectionState connecting
[WebRTC Plugin]: iceConnectionState checking
[WebRTC Plugin]: waiting ice connected
[WebRTC Plugin]: iceConnectionState completed
[WebRTC Plugin]: iceGatheringState complete
[WebRTC Plugin]: iceConnectionState connected
[WebRTC Plugin]: connectionState connected
[WebRTC Plugin]: ice transport { ipv4: true, isLocalNetwork: true, destinationId: '192.168.6.118' }
[WebRTC Plugin]: Connection is local network: true
[WebRTC Plugin]: done waiting ice connected
[WebRTC Plugin]: Error starting playback for WebRTC track. RPCResultError [TypeError]: Cannot read properties of undefined (reading 'allowBatteryPrebuffer')
[WebRTC Plugin]:     at V.ensurePrebufferSessions (/@scrypted/prebuffer-mixin/main.nodejs.js:1:131476)
[WebRTC Plugin]:     at async V.getVideoStream (/@scrypted/prebuffer-mixin/main.nodejs.js:1:128353)
[WebRTC Plugin]:     at async RpcPeer.handleMessageInternal (/server/node_modules/@scrypted/server/dist/rpc.js:591:37)
[WebRTC Plugin]: thread:main
[WebRTC Plugin]: @scrypted/prebuffer-mixin:host
[WebRTC Plugin]: host:@scrypted/webrtc {
[WebRTC Plugin]:   cause: undefined
[WebRTC Plugin]: }
[WebRTC Plugin]: signalingState closed
[WebRTC Plugin]: connectionState closed
[WebRTC Plugin]: iceConnectionState closed
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[WebRTC Plugin]: signalingState have-remote-offer
[WebRTC Plugin]: iceGatheringState gathering
[WebRTC Plugin]: signalingState stable
[WebRTC Plugin]: connectionState connecting
[WebRTC Plugin]: iceConnectionState checking
[WebRTC Plugin]: waiting ice connected
[WebRTC Plugin]: iceConnectionState completed
[WebRTC Plugin]: iceGatheringState complete
[WebRTC Plugin]: iceConnectionState connected
[WebRTC Plugin]: connectionState connected
[WebRTC Plugin]: ice transport { ipv4: true, isLocalNetwork: true, destinationId: '192.168.6.118' }
[WebRTC Plugin]: Connection is local network: true
[WebRTC Plugin]: done waiting ice connected
[WebRTC Plugin]: Error starting playback for WebRTC track. RPCResultError [TypeError]: Cannot read properties of undefined (reading 'allowBatteryPrebuffer')
[WebRTC Plugin]:     at V.ensurePrebufferSessions (/@scrypted/prebuffer-mixin/main.nodejs.js:1:131476)
[WebRTC Plugin]:     at async V.getVideoStream (/@scrypted/prebuffer-mixin/main.nodejs.js:1:128353)
[WebRTC Plugin]:     at async RpcPeer.handleMessageInternal (/server/node_modules/@scrypted/server/dist/rpc.js:591:37)
[WebRTC Plugin]: thread:main
[WebRTC Plugin]: @scrypted/prebuffer-mixin:host
[WebRTC Plugin]: host:@scrypted/webrtc {
[WebRTC Plugin]:   cause: undefined
[WebRTC Plugin]: }
[WebRTC Plugin]: signalingState closed
[WebRTC Plugin]: connectionState closed
[WebRTC Plugin]: iceConnectionState closed
########################
9/18/2023, 8:12:57 PM
########################
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[WebRTC Plugin]: signalingState have-remote-offer
[WebRTC Plugin]: iceGatheringState gathering
[WebRTC Plugin]: signalingState stable
[WebRTC Plugin]: connectionState connecting
[WebRTC Plugin]: iceConnectionState checking
[WebRTC Plugin]: waiting ice connected
[WebRTC Plugin]: iceConnectionState completed
[WebRTC Plugin]: iceGatheringState complete
[WebRTC Plugin]: iceConnectionState connected
[WebRTC Plugin]: connectionState connected
[WebRTC Plugin]: ice transport { ipv4: true, isLocalNetwork: true, destinationId: '192.168.6.118' }
[WebRTC Plugin]: Connection is local network: true
[WebRTC Plugin]: done waiting ice connected
[WebRTC Plugin]: Error starting playback for WebRTC track. RPCResultError [TypeError]: Cannot read properties of undefined (reading 'allowBatteryPrebuffer')
[WebRTC Plugin]:     at V.ensurePrebufferSessions (/@scrypted/prebuffer-mixin/main.nodejs.js:1:131476)
[WebRTC Plugin]:     at async V.getVideoStream (/@scrypted/prebuffer-mixin/main.nodejs.js:1:128353)
[WebRTC Plugin]:     at async RpcPeer.handleMessageInternal (/server/node_modules/@scrypted/server/dist/rpc.js:591:37)
[WebRTC Plugin]: thread:main
[WebRTC Plugin]: @scrypted/prebuffer-mixin:host
[WebRTC Plugin]: host:@scrypted/webrtc {
[WebRTC Plugin]:   cause: undefined
[WebRTC Plugin]: }
[WebRTC Plugin]: signalingState closed
[WebRTC Plugin]: connectionState closed
[WebRTC Plugin]: iceConnectionState closed
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[Snapshot Plugin]: creating error image with background Snapshot Failed false
########################
9/18/2023, 9:45:45 PM
########################
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
########################
9/18/2023, 10:01:45 PM
########################
[Rebroadcast Plugin]: prebuffer sessions releasing if started
[Rebroadcast Plugin]: prebuffer sessions starting in 5 seconds
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[Snapshot Plugin]: creating error image with background Snapshot Failed false
########################
9/18/2023, 10:11:16 PM
########################
[Rebroadcast Plugin]: error retrieving getVideoStreamOptions TypeError: Cannot read properties of undefined (reading 'map')
[Rebroadcast Plugin]:     at Object.onGet (/@scrypted/prebuffer-mixin/main.nodejs.js:1:151037)
[Rebroadcast Plugin]:     at async t.StorageSettings.getSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:48313)
[Rebroadcast Plugin]:     at async V.getMixinSettings (/@scrypted/prebuffer-mixin/main.nodejs.js:1:132916)
[Snapshot Plugin]: creating error image with background Snapshot Failed false
[WebRTC Plugin]: signalingState have-remote-offer
[WebRTC Plugin]: iceGatheringState gathering
[WebRTC Plugin]: signalingState stable
[WebRTC Plugin]: connectionState connecting
[WebRTC Plugin]: iceConnectionState checking
[WebRTC Plugin]: waiting ice connected
[WebRTC Plugin]: iceConnectionState completed
[WebRTC Plugin]: iceGatheringState complete
[WebRTC Plugin]: iceConnectionState connected
[WebRTC Plugin]: connectionState connected
[WebRTC Plugin]: ice transport { ipv4: true, isLocalNetwork: true, destinationId: '192.168.6.118' }
[WebRTC Plugin]: Connection is local network: true
[WebRTC Plugin]: done waiting ice connected
[WebRTC Plugin]: Error starting playback for WebRTC track. RPCResultError [TypeError]: Cannot read properties of undefined (reading 'allowBatteryPrebuffer')
[WebRTC Plugin]:     at V.ensurePrebufferSessions (/@scrypted/prebuffer-mixin/main.nodejs.js:1:131476)
[WebRTC Plugin]:     at async V.getVideoStream (/@scrypted/prebuffer-mixin/main.nodejs.js:1:128353)
[WebRTC Plugin]:     at async RpcPeer.handleMessageInternal (/server/node_modules/@scrypted/server/dist/rpc.js:591:37)
[WebRTC Plugin]: thread:main
[WebRTC Plugin]: @scrypted/prebuffer-mixin:host
[WebRTC Plugin]: host:@scrypted/webrtc {
[WebRTC Plugin]:   cause: undefined
[WebRTC Plugin]: }
[WebRTC Plugin]: signalingState closed
[WebRTC Plugin]: connectionState closed
[WebRTC Plugin]: iceConnectionState closed

**Comment by pilot1981** [comment]
any help?

**Comment by koush** [comment]
use ffmpeg parser maybe.

or use ffmpeg plugin.

**Comment by pilot1981** [comment]
solved using rebroadcast and opencv plugins
